lib/pure/asyncmacro.nim

```diff
@@ -440,7 +399,24 @@ macro multisync*(prc: untyped): untyped =
   result.add(asyncSingleProc(asyncPrc))
   result.add(sync)
 
-proc await*[T](x: T) =
-  ## The 'await' keyword is also defined here for technical
-  ## reasons. (Generic symbol lookup prepass.)
-  {.error: "Await only available within .async".}
+# seems simpler to define it here, than to define it in each multisync
+template await*(f: untyped): untyped =
+  when declared(internalRetFuture):
+    error "Await expects a Future"
```

Comment:
Please expand this:

```nim
error "Await expects a Future[T], got " & $typeof(f)
```

or similar.

Reply:
Ok, but i change all those errors to `await` because thats the name of it, `Await` might be incorrect